### ISSUE TITLE
Support for @-mentions in group forum posts

### ISSUE THREAD

**Issue by boonebgorges** [cuny-academic-commons/commons-in-a-box#378]
Here's an mu-plugin that does it: https://gist.github.com/boonebgorges/9db0df279b1bb57f8048b4cf3742307a

This needs cleanup to ensure that it works in the proper interfaces (group forum topics? replies? private messages? etc) and also to find a proper place to put the changes.

**Comment by r-a-y** [comment]
Yeah, there is code that is only specific to the Commons. Other than @-mentions support, it also:

 * Disables the "Image Caption" field from the TinyMCE Image button.
 * Adds a proper underline button to the TinyMCE and quicktags editor

**Comment by boonebgorges** [comment]
Good call, @r-a-y - I'll make sure to trim the fat when we merge.

**Comment by boonebgorges** [comment]
I've made an initial pass at this in 82e6838. It works, but only in the Visual tab. @r-a-y Is it supposed to work in the Text tab as well? I can't remember.

**Comment by r-a-y** [comment]
@boonebgorges - Yes, at-mentions are supposed to work on the "Text" tab.

BuddyPress checks the `'wp-editor-area'` selector to see if the mentions script should run: https://github.com/buddypress/buddypress/blob/3556968d91d981f945d656f795be00ea20234bd9/src/bp-activity/js/mentions.js#L244-L245.

However, OpenLab has a customization to remove this CSS class: https://github.com/cuny-academic-commons/openlab-theme/blob/55628697a917f70b52dbb5b698086026d37a43e1/lib/plugin-hooks.php#L145

I did a quick test and adding back the `'wp-editor-area'` CSS class fixes the issue.

**Comment by boonebgorges** [comment]
Amazing, thank you for tracking this down!

**Comment by bree-z** [comment]
This looks good, thanks!
